### PR TITLE
Remove deprecated config and reduce complexity

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
Remove the config because it's only really used for the dropwizard timelock server, and the values are actually hardcoded in our internal timelock to opposite values than the defaults, resulting in confusing and misleading code.
 
**Concerns (what feedback would you like?)**:
This is technically a devbreak for people using the dropwizard launcher as now we force them to use the same settings, so they might have to fix their configs.

### CODE REVIEW

File: timelock-server-benchmark-cluster/src/main/java/com/palantir/atlasdb/timelock/benchmarks/server/config/TimelockBenchmarkServerConfig.java [hunk 1]
```diff
@@ -41,4 +42,11 @@ public TimeLockRuntimeConfiguration runtime() {
         return runtime;
     }
 
+    public int threadPoolSize() {
+        return (int) (Math.max(256, 32 * Runtime.getRuntime().availableProcessors()) * 0.5);
```
Comment (felixdesouza) [inline timelock-server-benchmark-cluster/src/main/java/com/palantir/atlasdb/timelock/benchmarks/server/config/TimelockBenchmarkServerConfig.java]: what on earth? 😂 

Can we just do:

    Math.max(256, 32 / 2 * Runtime.getRuntime().availableProcessors())

File: timelock-server-benchmark-cluster/src/main/java/com/palantir/atlasdb/timelock/benchmarks/server/config/TimelockBenchmarkServerConfig.java [hunk 2]
```diff
@@ -41,4 +42,11 @@ public TimeLockRuntimeConfiguration runtime() {
...
+    }
+
+    public long blockTimeoutMs() {
+        return (long) (HumanReadableDuration.minutes(1).toMilliseconds() * 0.8);
```
Comment (felixdesouza) [inline timelock-server-benchmark-cluster/src/main/java/com/palantir/atlasdb/timelock/benchmarks/server/config/TimelockBenchmarkServerConfig.java]: I suppose

File: timelock-server-distribution/var/conf/timelock-oracle.yml [hunk 3]
```diff
@@ -1,32 +1,32 @@
-clients:
-  - test
-  - test2
-  - atlasete
-
-cluster:
-  localServer: localhost:8421
-  servers:
-    - localhost:8421
+install:
```
Comment (felixdesouza) [inline timelock-server-distribution/var/conf/timelock-oracle.yml]: does this even get run anywhere? taking into account oss considerations

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AbstractAsyncTimelockServiceIntegrationTest.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
 
     protected static final TestableTimelockCluster CLUSTER_WITH_ASYNC = new TestableTimelockCluster(
             LOCALHOST,
-            "paxosSingleServerWithAsyncLock.yml");
+            "paxosSingleServer.yml");
```
Comment (felixdesouza) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AbstractAsyncTimelockServiceIntegrationTest.java]: intentional?

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java [hunk 5]
```diff
@@ -86,11 +87,7 @@
     private static final List<String> CLIENTS = ImmutableList.of(CLIENT_1, CLIENT_2, CLIENT_3, LEARNER, ACCEPTOR);
     private static final String INVALID_CLIENT = "test2\b";
 
-    private static final int MAX_SERVER_THREADS = 100;
-    private static final int SELECTOR_THREADS = 8;
-    private static final int ACCEPTOR_THREADS = 4;
-    private static final int AVAILABLE_THREADS = MAX_SERVER_THREADS - SELECTOR_THREADS - ACCEPTOR_THREADS - 1;
-    private static final int SHARED_TC_LIMIT = AVAILABLE_THREADS;
+    private static final int SHARED_TC_LIMIT = CombinedTimeLockServerConfiguration.threadPoolSize();
```
Comment (felixdesouza) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java]: better name please

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java [hunk 6]
```diff
@@ -212,7 +209,8 @@ private int lockAndUnlockAndCountExceptions(List<LockService> lockServices, int
         AtomicInteger exceptionCounter = new AtomicInteger(0);
         futures.forEach(future -> {
             try {
-                assertNull(future.get());
+                LockRefreshToken test = future.get();
+                assertNull(test);
```
Comment (felixdesouza) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java]: `assertThat(test).isNull()`

there are like 2 instances each of `assertNull` `assertTrue` and `assertNotNull`, assertj-ize it

File: timelock-server/src/integTest/resources/paxosMultiServer1.yml [hunk 7]
```diff
@@ -1,23 +1,25 @@
...
+  cluster:
+    cluster:
+      security:
+        trustStorePath: "var/security/trustStore.jks"
+        trustStoreType: "JKS"
+        keyStorePath: "var/security/keyStore.jks"
+        keyStorePassword: "keystore"
+        keyStoreType: "JKS"
+      uris:
+      - "localhost:9081"
```
Comment (felixdesouza) [inline timelock-server/src/integTest/resources/paxosMultiServer1.yml]: nit: retain original order

File: timelock-server/src/integTest/resources/paxosMultiServer0.yml [hunk 8]
```diff
@@ -1,23 +1,25 @@
...
+  cluster:
+    cluster:
+      security:
+        trustStorePath: "var/security/trustStore.jks"
+        trustStoreType: "JKS"
+        keyStorePath: "var/security/keyStore.jks"
+        keyStorePassword: "keystore"
+        keyStoreType: "JKS"
+      uris:
+      - "localhost:9081"
```
Comment (felixdesouza) [inline timelock-server/src/integTest/resources/paxosMultiServer0.yml]: nit: original order

File: timelock-server/src/integTest/resources/paxosMultiServer2.yml [hunk 9]
```diff
@@ -1,23 +1,25 @@
...
+    cluster:
+      security:
+        trustStorePath: "var/security/trustStore.jks"
+        trustStoreType: "JKS"
+        keyStorePath: "var/security/keyStore.jks"
+        keyStorePassword: "keystore"
+        keyStoreType: "JKS"
+      uris:
+      - "localhost:9081"
+      - "localhost:9080"
```
Comment (felixdesouza) [inline timelock-server/src/integTest/resources/paxosMultiServer2.yml]: nit: original order

File: timelock-server/src/integTest/resources/paxosSingleServer.yml [hunk 10]
```diff
@@ -1,35 +1,31 @@
...
-  - acceptor
-
-useClientRequestLimit: true
+runtime:
+  paxos:
 
 server:
   minThreads: 1
-  maxThreads: 100
+  maxThreads: 200
```
Comment (felixdesouza) [inline timelock-server/src/integTest/resources/paxosSingleServer.yml]: intentional?

File: timelock-server/src/integTest/resources/paxosThreeServers.yml [hunk 11]
```diff
@@ -1,20 +1,24 @@
...
-
-cluster:
-  localServer: localhost:9060
-  servers:
-    - localhost:9060
-    - localhost:9061
-    - localhost:9062
+install:
+  paxos:
+    data-directory: "<TEMP_DATA_DIR>"
```
Comment (felixdesouza) [inline timelock-server/src/integTest/resources/paxosThreeServers.yml]: what's with this place holder?

File: timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/CombinedTimeLockServerConfiguration.java [hunk 12]
```diff
@@ -15,21 +15,37 @@
...
+    }
 
-    TimeLockRuntimeConfiguration runtime();
+    public static int threadPoolSize() {
+        return 128;
+    }
 
-    TimeLockDeprecatedConfiguration deprecated();
+    public static long blockingTimeoutMs() {
+        return (long) (HumanReadableDuration.minutes(1).toMilliseconds() * 0.8);
```
Comment (felixdesouza) [inline timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/CombinedTimeLockServerConfiguration.java]: isn't this elsewhere?

File: atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/EteSetup.java [hunk 13]
```diff
@@ -94,7 +94,7 @@ public static RuleChain setupCompositionWithTimelock(
             String composeFile,
             List<String> availableClientNames,
             Map<String, String> environment) {
-        waitDuration = Duration.TWO_MINUTES;
+        waitDuration = Duration.TEN_MINUTES;
```
Comment (felixdesouza) [inline atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/EteSetup.java]: intentional?

File: timelock-server/src/integTest/resources/paxosThreeServers.yml [hunk 14]
```diff
@@ -1,20 +1,24 @@
...
+  cluster:
+    cluster:
+      security:
+        trustStorePath: "var/security/trustStore.jks"
+        trustStoreType: "JKS"
+        keyStorePath: "var/security/keyStore.jks"
+        keyStorePassword: "keystore"
+        keyStoreType: "JKS"
+      uris:
+      - "localhost:9061"
```
Comment (felixdesouza) [inline timelock-server/src/integTest/resources/paxosThreeServers.yml]: nit: same order

**Comment by gmaretic** [comment]
@felixdesouza good to merge?

**Pull request merged**